src/s3.rs

```diff
@@ -104,14 +106,42 @@ pub fn list_objects(
         .try_flatten()
 }
 
+/// Retrieve an object from S3 as an `AsyncBufRead`.
+///
+/// # Example
+///
+/// ```no_run
+/// use aws_config;
+/// use cobalt_aws::s3::{get_client, async_get_object};
+/// use futures::AsyncReadExt;
+///
+/// # tokio_test::block_on(async {
+/// let shared_config = aws_config::load_from_env().await;
+/// let client = get_client(&shared_config).unwrap();
+/// let mut reader = async_get_object(&client, "my-bucket", "my-key").await.unwrap();
+/// let mut buffer = String::new();
+/// reader.read_to_string(&mut buffer).await.unwrap();
+/// println!("{}", buffer);
+/// # })
+/// ```
+pub async fn async_get_object(
```

Comment:
High-level question about the name here: do we expect to ever have a non-`async_` version of this that returns some other kind of abstraction? I'm wondering if it would make sense for it to just be the canonical `get_object` exposed by this module.

Reply:
Yeah, good question, I was just experimenting with using this in context and I think you're right, the `async_` prefix isn't helping here, and is possibly hindering. I'll change it to just `get_object` in this PR and we can iterate from there 👍